amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/MultiStreamLease.java

```diff
@@ -0,0 +1,83 @@
+package software.amazon.kinesis.leases;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+import org.apache.commons.lang3.Validate;
+
+import java.util.Objects;
+
+import static com.google.common.base.Verify.verifyNotNull;
+
+@Setter
+@NoArgsConstructor
+@Getter
+@Accessors(fluent = true)
+public class MultiStreamLease extends Lease {
+
+    @NonNull private String streamIdentifier;
+    @NonNull private String shardId;
+
+    public MultiStreamLease(MultiStreamLease other) {
+        super(other);
+        streamIdentifier(other.streamIdentifier);
+        shardId(other.shardId);
+    }
+
+    @Override
+    public void update(Lease other) {
+        MultiStreamLease casted = validateAndCast(other);
+        super.update(casted);
+        streamIdentifier(casted.streamIdentifier);
+        shardId(casted.shardId);
+    }
+
+    public static String getLeaseKey(String streamIdentifier, String shardId) {
+        verifyNotNull(streamIdentifier, "streamIdentifier should not be null");
+        verifyNotNull(shardId, "shardId should not be null");
+        return streamIdentifier + ":" + shardId;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(super.hashCode(), streamIdentifier);
+    }
```

Comment:
Dont we need shard id in the Equals and Hashcode comparison ?

Reply:
Actually the shardId info is present in the base class leaseKey. Will refactor as required